bundles/org.openhab.binding.mqtt.awtrixlight/src/main/java/org/openhab/binding/mqtt/awtrixlight/internal/action/AwtrixActions.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.mqtt.awtrixlight.internal.action;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.mqtt.awtrixlight.internal.handler.AwtrixLightBridgeHandler;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+
+/**
+ * Actions for the Awtrix clock.
+ *
+ * @author Thomas Lauterbach - Initial contribution
+ */
+@ThingActionsScope(name = "mqtt.awtrixlight")
+@NonNullByDefault
+public class AwtrixActions implements ThingActions {
+
+    private @Nullable AwtrixLightBridgeHandler handler;
+
+    @Override
+    public void setThingHandler(ThingHandler handler) {
+        this.handler = (AwtrixLightBridgeHandler) handler;
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return handler;
+    }
+
+    @RuleAction(label = "Blink Indicator", description = "Blink indicator with indicatorId")
+    public void blinkIndicator(int indicatorId, int[] rgb, int blinkTimeInMs) {
+        AwtrixLightBridgeHandler localHandler = this.handler;
+        if (localHandler != null) {
+            localHandler.blinkIndicator(indicatorId, rgb, blinkTimeInMs);
+        }
+    }
+
+    public static void blinkIndicator(@Nullable ThingActions actions, int indicatorId, int[] rgb, int blinkTimeInMs) {
+        if (actions instanceof AwtrixActions) {
+            ((AwtrixActions) actions).blinkIndicator(indicatorId, rgb, blinkTimeInMs);
+        } else {
+            throw new IllegalArgumentException("Instance is not an AwtrixActions class.");
+        }
+    }
+
+    @RuleAction(label = "Fade Indicator", description = "Fade indicator with indicatorId")
+    public void fadeIndicator(int indicatorId, int[] rgb, int fadeTimeInMs) {
+        AwtrixLightBridgeHandler localHandler = this.handler;
+        if (localHandler != null) {
+            localHandler.fadeIndicator(indicatorId, rgb, fadeTimeInMs);
+        }
+    }
+
+    public static void fadeIndicator(@Nullable ThingActions actions, int indicatorId, int[] rgb, int fadeTimeInMs) {
+        if (actions instanceof AwtrixActions) {
+            ((AwtrixActions) actions).fadeIndicator(indicatorId, rgb, fadeTimeInMs);
```

Comment:
Please fix all similar occurences in this file.
```suggestion
        if (actions instanceof AwtrixActions awtrixActions) {
            awtrixActions.fadeIndicator(indicatorId, rgb, fadeTimeInMs);
```

Reply:
Done